### PR TITLE
🌱 Bump Go to v1.22.2

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Bumps Go to v1.22.2

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Part of #10026

/area dependency

### CODE REVIEW

File: cloudbuild-nightly.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ options:
   substitution_option: ALLOW_LOOSE
   machineType: 'E2_HIGHCPU_8'
 steps:
-  - name: 'gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20230522-312425ae46'
+  - name: 'gcr.io/k8s-staging-test-infra/gcb-docker-gcloud@sha256:de53ba7cd20326776a00adb065430a8bb51beaf24876ffcbd4e8f71b74dbc22d' # v20240210-29014a6e3a
```
Comment (killianmuldoon) [inline cloudbuild-nightly.yaml:8]: Should this generally be updated with the go version?

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

one question

File: controlplane/kubeadm/internal/webhooks/scale.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ func (v *ScaleValidator) SetupWebhookWithManager(mgr ctrl.Manager) error {
 // ScaleValidator validates KCP for replicas.
 type ScaleValidator struct {
 	Client  client.Reader
-	decoder *admission.Decoder
+	decoder admission.Decoder
```
Comment (sbueringer) [inline controlplane/kubeadm/internal/webhooks/scale.go:48]: Should be reverted

File: netlify.toml [hunk 3]
```diff
@@ -4,7 +4,7 @@
     publish = "docs/book/book"
 
 [build.environment]
-    GO_VERSION = "1.21"
+    GO_VERSION = "1.22.2"
```
Comment (sbueringer) [inline netlify.toml:7]: If possible I would only pin the minor version

(but maybe it's not possible anymore, then it's fine)

File: cloudbuild-nightly.yaml (hunk 1 above)
Comment (sbueringer) [reply cloudbuild-nightly.yaml:8]: I think we have to use a version that contains 1.22 if that's what you're asking

File: netlify.toml (hunk 3 above)
Comment (chrischdi) [reply netlify.toml:7]: that's not possible, it would fail...

**Pull request merged**